src/systems/a380_systems/src/pneumatic.rs

```diff
@@ -2390,42 +2531,65 @@ mod tests {
         let test_bed = test_bed_with()
             .stop_eng1()
             .stop_eng2()
+            .stop_eng3()
+            .stop_eng4()
             .in_isa_atmosphere(altitude)
             .mach_number(MachNumber(0.))
             .set_pack_flow_pb_is_auto(1, false)
             .set_pack_flow_pb_is_auto(2, false)
             .and_stabilize();
 
-        assert!((test_bed.ip_pressure(1) - ambient_pressure).abs() < pressure_tolerance());
-        assert!((test_bed.ip_pressure(2) - ambient_pressure).abs() < pressure_tolerance());
-
-        assert!((test_bed.hp_pressure(1) - ambient_pressure).abs() < pressure_tolerance());
-        assert!((test_bed.hp_pressure(2) - ambient_pressure).abs() < pressure_tolerance());
-
-        assert!((test_bed.transfer_pressure(1) - ambient_pressure).abs() < pressure_tolerance());
-        assert!((test_bed.transfer_pressure(2) - ambient_pressure).abs() < pressure_tolerance());
-
-        assert!(
-            (test_bed.precooler_inlet_pressure(1) - ambient_pressure).abs() < pressure_tolerance()
-        );
-        assert!(
-            (test_bed.precooler_inlet_pressure(2) - ambient_pressure).abs() < pressure_tolerance()
-        );
-
-        assert!(
-            (test_bed.precooler_outlet_pressure(1) - ambient_pressure).abs() < pressure_tolerance()
-        );
-        assert!(
-            (test_bed.precooler_outlet_pressure(2) - ambient_pressure).abs() < pressure_tolerance()
-        );
-
-        assert!(!test_bed.hp_valve_is_open(1));
-        assert!(!test_bed.hp_valve_is_open(2));
+        for engine_number in 1..=4 {
+            assert!(
+                (test_bed.ip_pressure(engine_number) - ambient_pressure).abs()
+                    < pressure_tolerance()
+            );
```

Comment:
You could do the following:
```suggestion
            assert_about_eq!(
                test_bed.ip_pressure(engine_number).get::<pascal>(),
                ambient_pressure.get::<pascal>(),
                pressure_tolerance().get::<pascal>()
            );
```
But the inconvenience is that you need to extract the raw values.